pkg/apis/cluster/v1alpha1/common_types.go

```diff
@@ -39,5 +42,25 @@ type ProviderConfig struct {
 // ProviderConfigSource represents a source for the provider-specific
 // resource configuration.
 type ProviderConfigSource struct {
```

Comment:
mvladev: https://github.com/kubernetes/kube-deploy/pull/659/files#r177866614

roberthbailey: I don't think it should be necessary to specify the capacity / allocatable on a single machine if you inline the provider config. I think of the class is a separate way to bring the provider specific data, not something inherent to the spec of the machine.

Reply:
/cc @mvladev - additional thoughts on this?

Reply:
@roberthbailey does that mean machine class is just a collection of provider specific configuration? 
Advantages of machine classes:
* less configuration lines in the machine objects by keeping common bits in a machine class object

Any other?

Reply:
> less configuration lines in the machine objects by keeping common bits in a machine class object
- more precisely, externalize the `providerConfig` to separate objects.
- as discussed in wg-call, it could also be used for defaulting/versioning via apiserver-webhooks, which could be hard to achieve in only inlined.